### PR TITLE
Remove remaining directory server mentions

### PR DESCRIPTION
**Pull request by pljones**
**Short description of changes**

This is a "Work in progress" -- on the basis I'm not 100% sure I've found all reference _still_!

CHANGELOG: Further clean up source references to "directory server" to current terminology

**Context: Fixes an issue?**

We now refer to "directory server" as "Directory" in the documentation.  This change tries to find more places the software gets it wrong and correct them.

**Does this change need documentation? What needs to be documented and how?**

Yes - the help string changes, as well as more translations being added.

**Status of this Pull Request**

I've run in client, server and directory modes successfully.

**What is missing until this pull request can be merged?**

Works for me ... but yes, checking the changes - and impact on translators especially - would be useful.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: docs/JSON-RPC.md [hunk 1]
```diff
@@ -280,7 +280,7 @@ Results:
 | result.city | string | The server city. |
 | result.countryId | number | The server country ID (see QLocale::Country). |
 | result.welcomeMessage | string | The server welcome message. |
-| result.directoryServer | string | The directory server to which this server requested registration, or blank if none. |
+| result.directoryServer | string | The directory to which this server requested registration, or blank if none. |
```
Comment (mcfnord) [inline docs/JSON-RPC.md:283]:
```suggestion
| result.directoryServer | string | The directory where this server requested registration, or blank if none. |
```

I don't know if this is better. _To which_ is just a bit... _witchy_.

File: src/connectdlg.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ void CConnectDlg::RequestServerList()
     }
     cbxDirectoryServer->blockSignals ( false );
 
-    // Get the IP address of the directory server (using the ParseNetworAddress
+    // Get the IP address of the directory (using the ParseNetworAddress
```
Comment (mcfnord) [inline src/connectdlg.cpp:240]:
```suggestion
    // Get the IP address of the directory (using the ParseNetworkAddress
```

I have not checked if the function name does omit the k.

File: docs/JSON-RPC.md (hunk 1 above)
Comment (pljones) [reply docs/JSON-RPC.md:283]: I want to imply a request being sent _from_ the server _to_ the directory.  So the "to" helps, I think.  That then limits the rest of the phrase...

File: src/connectdlg.cpp (hunk 2 above)
Comment (pljones) [reply src/connectdlg.cpp:240]: Good spot, if a bit late ;)

File: docs/JSON-RPC.md (hunk 1 above)
Comment (mcfnord) [reply docs/JSON-RPC.md:283]: Albatross, English! _where this server requested registration_ has no trouble indicating the actor.

**Pull request closed without merging**